core/src/main/java/org/jruby/ir/targets/simple/NormalValueCompiler.java

```diff
@@ -125,6 +127,16 @@ public void pushByteList(final ByteList bl) {
         });
     }
 
+    public void pushRange(Runnable begin, Runnable end, boolean exclusive) {
+        cacheValuePermanentlyLoadContext("range", RubyRange.class, null, () -> {
+            compiler.loadContext();
+            begin.run();
+            end.run();
+            compiler.adapter.pushBoolean(exclusive);
+            compiler.adapter.invokestatic(p(RubyRange.class), "newRegexp", sig(RubyRange.class, ThreadContext.class, IRubyObject.class, IRubyObject.class, boolean.class));
```

Comment:
the RubyRange factory method name (`newRegexp`) seems like a copy pasta typo

Reply:
And here I was thinking "huh, maybe we need a better way to test the non-indy logic for constant values". Will fix and add a way to test.

Reply:
Adding an AOT mode to the compiler's `general_spec.rb` would have found this:

```
  10) JRuby's AOTSpecUtils assigns literal values to locals
      Failure/Error: Unable to infer file and line number from backtrace
      
      Java::JavaLang::NoSuchMethodError:
        'org.jruby.RubyRange org.jruby.RubyRange.newRegexp(org.jruby.runtime.ThreadContext, org.jruby.runtime.builtin.IRubyObject, org.jruby.runtime.builtin.IRubyObject, boolean)'
      # Users.headius.projects.jruby.spec.compiler.general_spec.range0(Users/headius/projects/jruby/spec/compiler//Users/headius/projects/jruby/spec/compiler/general_spec.rb)
      # Users.headius.projects.jruby.spec.compiler.general_spec.<main>(/Users/headius/projects/jruby/spec/compiler/general_spec.rb:170)
      # jdk.internal.reflect.GeneratedMethodAccessor22.invoke(jdk/internal/reflect/GeneratedMethodAccessor22)
      # jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(jdk/internal/reflect/DelegatingMethodAccessorImpl.java:43)
      # java.lang.reflect.Method.invoke(java/lang/reflect/Method.java:566)
      # RUBY.compile_run(/Users/headius/projects/jruby/spec/compiler/general_spec.rb:129)
      # RUBY.run(/Users/headius/projects/jruby/spec/compiler/general_spec.rb:74)
      # RUBY.<main>(/Users/headius/projects/jruby/spec/compiler/general_spec.rb:170)
```